### PR TITLE
Add support for patch_base_image when not running locally

### PR DESCRIPTION
**Pull request by jprokos26**
Addresses #1930 

Searches for `patch_base_image` in more than one place:
1. Check if file exists in module path
2. Check if file exists in `armory.paths.runtime_paths().cwd`
3. Check if supplied `patch_base_image` is actually a url to an image we can download
Else fail

### CODE REVIEW

**Comment by jprokos26** [comment]
Whenever I tried to `pip install -e .` with these changes and move to a different directory when I launch the docker container the armory being run does not have these changes so I was unable to test that the cwd gets mapped properly to `runtime_paths().cwd` when running as a pip installed package not locally.

File: armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py [hunk 1]
```diff
@@ -48,12 +49,34 @@ def create_initial_image(self, size, hsv_lower_bound, hsv_upper_bound):
...
+        if not os.path.exists(patch_base_image_path):
+            patch_base_image_path = os.path.abspath(
+                os.path.join(paths.runtime_paths().cwd, self.patch_base_image)
+            )
+        # image not in cwd or module, check if it is a url to an image
+        if not os.path.exists(patch_base_image_path):
+            import requests
+
+            try:
+                response = requests.get(self.patch_base_image, allow_redirects=True)
```
Comment (jprokos26) [inline armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py:67]: Any worry about safety of links? i.e. zipbombs or something?

Comment (mwartell) [reply armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py:67]: cf. https://stackoverflow.com/questions/14270698/get-file-size-using-python-requests-while-only-getting-the-header for sample usage

Comment (jprokos26) [reply armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py]: Resolved with https://github.com/twosixlabs/armory/pull/1939/commits/67e002049b4c5c5a03f7b1074231c5346da52ae7

**Review by christopherwoodall — APPROVED** [review]
LGTM!

**Pull request merged**